packages/core/assistant-testing/src/functions/research/research.test.ts

```diff
@@ -56,17 +55,18 @@ const TestLayer = Layer.mergeAll(
         types: [...ResearchDataTypes, ResearchGraph, Blueprint.Blueprint],
       }),
       CredentialsService.configuredLayer([{ service: 'exa.ai', apiKey: EXA_API_KEY }]),
-      LocalFunctionExecutionService.layer,
-      RemoteFunctionExecutionService.mockLayer,
-      TracingService.layerNoop,
+      FunctionInvocationService.layerTest({ functions: [research] }).pipe(
+        Layer.provideMerge(ComputeEventLogger.layerFromTracing),
+        Layer.provideMerge(TracingService.layerNoop),
+      ),
     ),
   ),
 );
 
 describe('Research', { timeout: 600_000 }, () => {
   it.effect(
     'call a function to generate a research report',
-    Effect.fnUntraced(
+    Effect.fn(
```

Comment:
When using Effect.fn provide the name `Effect.fn('myFunc')(function* ({}) { ... })`.
But here you are in a test so you don't need Effect.fn -- use `Effect.fnUntraced(function* () { .. })`

NOTE: There's a type-inference issue with functions that ignore their parameters. I found that in tests putting `{}` as the parameter pattern works around the issue:

```ts
Effect.fnUntraced(function* ({}) { // <---- no error

})
```